.dev/compare_branches.R

```diff
@@ -146,7 +146,11 @@ test_encoding <- function(dir) {
   tryCatch({
     lapply(
       list.files(dir, pattern = "(?i)\\.r(?:md)?$", recursive = TRUE, full.names = TRUE),
-      function(x) nchar(readLines(x, warn = FALSE))
+      function(x) {
+        con <- file(x, encoding = lintr:::find_default_encoding(x))
+        on.exit(close(con))
+        nchar(readLines(con, warn = FALSE))
```

Comment:
nice! purely out of curiosity, any sense for how much this increases our "hit rate" for parsing packages correctly?

Reply:
I'd expect basically 100% since all Packages must provide a DESCRIPTION file and
Writing R Extensions says

> If the `DESCRIPTION' file is not entirely in ASCII it should contain an `Encoding' field specifying an encoding. (... also used for R files ...)

Reply:
I did think I found a package that failed on current HEAD but didn't have an Encoding listed. I might be misremembering. No matter :)